packages/fx-core/src/plugins/resource/bot/question.ts

```diff
@@ -171,6 +153,29 @@ export const showNotificationTriggerCondition = {
   containsAny: [NotificationOptionItem.id],
 };
 
+export function getNotificationTriggerQuestionCondition(runtime: Runtime) {
+  return {
+    validFunc: async (input: unknown, inputs?: Inputs) => {
+      if (inputs?.platform === Platform.CLI) {
+        if (isCLIDotNetEnabled()) {
+          if (inputs && inputs[CoreQuestionNames.Runtime] === runtime) {
+            return undefined;
+          }
+        } else if (runtime == Runtime.Node) {
+          return undefined;
+        }
+      }
+      if (inputs?.platform === Platform.VS && runtime === Runtime.Dotnet) {
+        return undefined;
+      }
+      if (inputs?.platform === Platform.VSCode && runtime == Runtime.Node) {
```

Comment:
Same here. Maybe we can consider if we need a map to describe the decision tree, instead of a lot of if condition. The logic looks like, when isCLIDotNetEnabled() is turn off, there is a mapping (Platform -> Runtime) that
```
Platform.CLI -> Runtime.Node
Platform.VS-> Runtime.Dotnet
Platform.VSCode-> Runtime.Node
```
So, the logic is to return undefined when ```runtime === getMap(inputs?.platform)```, else return ```runtime is not ${runtime}```